src/Access/Credentials.h

```diff
@@ -0,0 +1,55 @@
+#pragma once
+
+#include <common/types.h>
+#include <memory>
+
+
+namespace DB
+{
+
+class Credentials
```

Comment:
I still don't understand what's the purpose of this class and why we can't replace this class with a simple structure like this

```
stuct Credentials
{
  String user_name;
  String password;
  bool check_password = true;
};
```
All logic can be done inside the `Authentication` class.

Reply:
There are different kinds of `Credentials`. Some don't contain a password, others have some state that indicate that the check has been performed already (Kerberos.)